pkg/aws/ec2/api/wrapper.go

```diff
@@ -639,6 +656,38 @@ func (e *ec2Wrapper) DescribeNetworkInterfaces(input *ec2.DescribeNetworkInterfa
 	return describeNetworkInterfacesOutput, err
 }
 
+// DescribeNetworkInterfacesPages returns network interfaces that match the filters specified in the input with MaxResult set to 1000(max value)
+// This API is used during periodic ENI cleanup routine and trunk initialization to list all network interfaces that match the given filters (vpc-id or subnet-id, and tag)
+// Only required fields, network interface ID and tag set, is populated to avoid consuming extra memory
+func (e *ec2Wrapper) DescribeNetworkInterfacesPages(input *ec2.DescribeNetworkInterfacesInput) ([]*ec2.NetworkInterface, error) {
```

Comment:
Why do you need retrun a slice of NetworkInterface instead of a slice of string (ENI ID)? Since I see in `append` you just populate ENI ID and TagSet to every NetworkInterface, I am not sure why you need `TagSet`?